bluemira/base/reactor_config.py

```diff
@@ -229,12 +226,51 @@ def _check_args_are_strings(self, args: Iterable[str]):
             if not isinstance(a, str):
                 raise ReactorConfigError("args must be strings")
 
+    def _expand_paths_in_dict(self, d: Dict[str, Any], rel_path: Path):
+        """
+        Expand all file paths in the dict
+        by replacing their values with the json file contents.
+
+        This mutates the passed in dict.
+        """
+        for k in d:
+            d[k], rel_path_from = self._extract_and_expand_file_data_if_needed(
+                d[k], rel_path
+            )
+            if isinstance(d[k], dict):
+                self._expand_paths_in_dict(d[k], rel_path_from)
+
+    def _extract_and_expand_file_data_if_needed(
+        self, value: Any, rel_path: Path
+    ) -> Tuple[Union[Any, dict], str]:
+        if not isinstance(value, str):
+            return value, rel_path
+        if not value.startswith(_FILEPATH_PREFIX):
+            return value, rel_path
+
+        # remove _FILEPATH_PREFIX
+        f_path = value[len(_FILEPATH_PREFIX) :]
+
+        # check if rel path
```

Comment:
Nitpicky, but it might add some value to this comment to say what the path is relative to. I guess relative to the directory of the config file that the path appears in? Putting myself in the shoes of someone who's trying to work out what the code is doing with no prior knowledge, I guess it's just not 100% clear here what the `rel_path` variable actually is.